SIL.WritingSystems/Sldr.cs

```diff
@@ -470,55 +471,90 @@ private static void LoadLanguageTagsIfNecessary()
 				CreateSldrCacheDirectory();
 
 				cachedAllTagsPath = Path.Combine(SldrCachePath, "langtags.json");
+				string etagPath;
+				etagPath = Path.Combine(SldrCachePath, "langtags.json.etag");
 				var sinceTime = _embeddedAllTagsTime.ToUniversalTime();
 				if (File.Exists(cachedAllTagsPath))
 				{
 					var fileTime = File.GetLastWriteTimeUtc(cachedAllTagsPath);
 					if (sinceTime > fileTime)
+					{
 						// delete the old langtags.json file if a newer embedded one is available.
 						// this can happen if the application is upgraded to use a newer version of SIL.WritingSystems
 						// that has an updated embedded langtags.json file.
 						File.Delete(cachedAllTagsPath);
+						File.Delete(etagPath);
+					}
 					else
 						sinceTime = fileTime;
 				}
-				sinceTime += TimeSpan.FromSeconds(1);
-				try
-				{
-					if (_offlineMode)
-						throw new WebException("Test mode: SLDR offline so accessing cache", WebExceptionStatus.ConnectFailure);
 
+			}
+			_languageTags = new ReadOnlyKeyedCollection<string, SldrLanguageTagInfo>(ParseAllTagsJson(cachedAllTagsPath));
+		}
 
-					// get SLDR langtags.json from the SLDR api compressed
-					// it will throw WebException or have status HttpStatusCode.NotModified if file is unchanged or not get it
-					var langtagsUrl =
-						$"{SldrRepository}index.html?query=langtags&ext=json{StagingParameter}";
-					var webRequest = (HttpWebRequest) WebRequest.Create(Uri.EscapeUriString(langtagsUrl));
-					webRequest.UserAgent = UserAgent;
-					webRequest.IfModifiedSince = sinceTime;
-					webRequest.Timeout = 10000;
-					webRequest.AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate;
-					using var webResponse = (HttpWebResponse) webRequest.GetResponse();
-					if (webResponse.StatusCode != HttpStatusCode.NotModified)
+		public static void DownloadLanguageTags()
+		{
+			CreateSldrCacheDirectory();
+			if (_offlineMode)
+					throw new WebException("Test mode: SLDR offline so accessing cache", WebExceptionStatus.ConnectFailure);
+
+			string cachedAllTagsPath;
+			cachedAllTagsPath = Path.Combine(SldrCachePath, "langtags.json");
+			string etagPath;
+			etagPath = Path.Combine(SldrCachePath, "langtags.json.etag");
+			string etag;
+			string currentEtag;
+			try
+			{
+				// get SLDR langtags.json from the SLDR api compressed
+				// it will throw WebException or have status HttpStatusCode.NotModified if file is unchanged or not get it
+				var langtagsUrl =
+					$"{SldrRepository}index.html?query=langtags&ext=json{StagingParameter}";
+				var webRequest = (HttpWebRequest)WebRequest.Create(Uri.EscapeUriString(langtagsUrl));
+				webRequest.Headers.Set("If-None-Match", etag);
+				webRequest.UserAgent = UserAgent;
+				webRequest.Timeout = 10000;
+				webRequest.AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate;
+				using var webResponse = (HttpWebResponse)webRequest.GetResponse();
+				if (webResponse.StatusCode != "OK")
```

Comment:
`StatusCode` returns an enum, so you can't compare to `string`:

```suggestion
				if (webResponse.StatusCode != HttpStatusCode.OK)
```